backend/webapp/src/test/java/org/bonitasoft/web/designer/model/asset/AssetTest.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.model.asset;
+
+
+import static org.bonitasoft.web.designer.builder.AssetBuilder.aFilledAsset;
+
+import org.bonitasoft.web.designer.config.DesignerConfig;
+import org.bonitasoft.web.designer.model.page.Page;
+import org.bonitasoft.web.designer.repository.BeanValidator;
+import org.bonitasoft.web.designer.repository.exception.ConstraintValidationException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class AssetTest {
+
+    private BeanValidator beanValidator = new DesignerConfig().beanValidator();
+
+    private Asset<Page> asset;
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void init() {
+        asset = aFilledAsset();
+
+    }
+
+    @Test
+    public void should_be_valid() {
+        beanValidator.validate(asset);
+    }
+
+    @Test
+    public void should_be_invalid_when_name_is_null() {
+        exception.expect(ConstraintValidationException.class);
+        exception.expectMessage("Asset name should not be blank");
+        asset.setName(null);
+
+        beanValidator.validate(asset);
+    }
+
+    @Test
+    public void should_be_invalid_when_name_containing_space() {
+        exception.expect(ConstraintValidationException.class);
+        exception.expectMessage("Asset name should contains only alphanumeric characters, with no space, and the file extension in lower case");
+        asset.setName("my name.js");
+
+        beanValidator.validate(asset);
+    }
+
+
+    @Test
+    public void should_be_invalid_when_name_containing_special_characters() {
+        exception.expect(ConstraintValidationException.class);
+        exception.expectMessage("Asset name should contains only alphanumeric characters, with no space, and the file extension in lower case");
+        asset.setName("myé&name.js");
+
+        beanValidator.validate(asset);
+    }
+
+    @Test
+    public void should_be_invalid_when_extension_uppercase() {
```

Comment:
Don't think we need to do that, will be painfull for images that have often uppercase extensions toto.PNG